include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -310,14 +316,17 @@ __pattern_find_end(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _
         return __res ? 0 : __rng1.size();
     }
 
+    sycl::queue __q_local = __exec.queue();
+
     using _Predicate = unseq_backend::multiple_match_pred<_Pred>;
     using _TagType = __par_backend_hetero::__parallel_find_backward_tag<_Range1>;
 
-    return oneapi::dpl::__par_backend_hetero::__parallel_find_or(
-        _BackendTag{},
-        __par_backend_hetero::make_wrapped_policy<__par_backend_hetero::__find_policy_wrapper>(
-            ::std::forward<_ExecutionPolicy>(__exec)),
-        _Predicate{__pred}, _TagType{}, ::std::forward<_Range1>(__rng1), ::std::forward<_Range2>(__rng2));
+    using _CustomName = oneapi::dpl::__internal::__policy_kernel_name<_ExecutionPolicy>;
+
+    return oneapi::dpl::__par_backend_hetero::__parallel_find_or<
+        __par_backend_hetero::__find_policy_wrapper<_CustomName>>(_BackendTag{}, __q_local, _Predicate{__pred},
+                                                                  _TagType{}, ::std::forward<_Range1>(__rng1),
+                                                                  ::std::forward<_Range2>(__rng2));
```

Comment:
I recommend doing another pass and replacing `::std` with `std` in code regions that are touched such as this one.

Reply:
I think I have two options:
1) replace `::std` to `std` in all modified files
2) Do not change this at all due we haven't essential necessity for that.
What is the best from your point of view?

Reply:
Also I think that due large amount of changes in this PR option(1) will make it difficult to review the real changes

Reply:
It's not a big deal from my side, so it's your decision. If I recall, what we decided upon a while ago was to try to change the code gradually as we touch old code that has `::std`. We did not want to make a large commit since that would muddle the commit history / commit blaming.

Probably, what would be nice in the future is if we had some script that could do this for touched changes, so we don't waste time trying to manually correct it all ourselves.